vars/githubPublishRelease.groovy

```diff
@@ -0,0 +1,133 @@
+import com.sap.piper.Utils
+import com.sap.piper.ConfigurationHelper
+
+import groovy.transform.Field
+
+@Field String STEP_NAME = 'githubPublishRelease'
+@Field Set GENERAL_CONFIG_KEYS = ['githubApiUrl', 'githubTokenCredentialsId', 'githubServerUrl']
+@Field Set STEP_CONFIG_KEYS = [
+    'addClosedIssues',
+    'addDeltaToLastRelease',
+    'customFilterExtension',
+    'excludeLabels',
+    'githubApiUrl',
+    'githubTokenCredentialsId',
+    'githubOrg',
+    'githubRepo',
+    'githubServerUrl',
+    'releaseBodyHeader',
+    'version'
+]
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+        def script = parameters.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        // load default & individual configuration
+        Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .addIfEmpty('githubOrg', script.commonPipelineEnvironment.getGithubOrg())
+            .addIfEmpty('githubRepo', script.commonPipelineEnvironment.getGithubRepo())
+            .addIfEmpty('version', script.commonPipelineEnvironment.getArtifactVersion())
+            .withMandatoryProperty('githubOrg')
+            .withMandatoryProperty('githubRepo')
+            .withMandatoryProperty('githubTokenCredentialsId')
+            .withMandatoryProperty('version')
+            .use()
+
+        new Utils().pushToSWA([step: STEP_NAME], config)
+
+        withCredentials([string(credentialsId: config.githubTokenCredentialsId, variable: 'TOKEN')]) {
+            def releaseBody = config.releaseBodyHeader?"${config.releaseBodyHeader}<br />":''
+            def content = getLastRelease(config, TOKEN)
+            if (config.addClosedIssues)
+                releaseBody += addClosedIssue(config, TOKEN, content.published_at)
+            if (config.addDeltaToLastRelease)
+                releaseBody += addDeltaToLastRelease(config, content.tag_name)
+            postNewRelease(config, TOKEN, releaseBody)
+        }
+    }
+}
+
+Map getLastRelease(config, TOKEN){
+    def result = [:]
+    try {
+        def response = httpRequest "${config.githubApiUrl}/repos/${config.githubOrg}/${config.githubRepo}/releases/latest?access_token=${TOKEN}"
+        result = readJSON text: response.content
+    } catch (e) {
+        echo "[${STEP_NAME}] This is the first release - no previous releases available ${e.getMessage()}"
+        config.addDeltaToLastRelease = false
```

Comment:
[nit] IMO we should avoid the habit of updating the config outside the config map. But maybe this is only this it is too much a personal habit and does not fit well for some kind of a project habit. In case we consider it as a benefit at some later point in time to have an immutable config map we can still solve that at that point in time.